<launch>

    <arg name="rate" default="100.0"/>
    <arg name="prefix" default=""/>
    
    <param name="robot_description" 
        textfile="$(env ROBOTOLOGY_ROOT)/robots/iit-cogimon-ros-pkg/cogimon_urdf/urdf/cogimon.urdf"/>
    
    <param name="robot_description_semantic"
        textfile="$(env ROBOTOLOGY_ROOT)/robots/iit-cogimon-ros-pkg/cogimon_srdf/srdf/cogimon.srdf"/>
    
    <param name="cartesian/problem_description" 
        textfile="$(find centroidal_planner)/cartesio_stack/coman_stack.yaml"/>
        
    <include file="$(find cartesian_interface)/launch/cartesio.launch">
        <arg name="rate" value="$(arg rate)"/>
        <arg name="prefix" value="$(arg prefix)"/>
    </include>    


    <rosparam command="load" param="cartesian/torque_offset" file="$(env ROBOTOLOGY_ROOT)/build/external/xbot_ros/tau_offset.yaml"/>

     <node pkg="cartesian_interface" 
          type="force_estimation_node" 
          name="force_estimator"
          output="screen">
          
          
        <remap from="~torque_offset" to="cartesian/torque_offset"/>
          
        <param name="is_model_floating_base" value="true"/>
        <param name="model_type" value="RBDL"/>
        
        <param name="links" type="yaml" value="[l_ball_tip, r_ball_tip]"/>
        <param name="chains" type="yaml" value="[left_arm, right_arm]"/>
        
    </node>
    
                
    <node pkg="centroidal_planner"
          type="forza_giusta_node"
          name="forza_giusta_node_torque_ffwd" 
          output="screen">
          
          <arg name="rate" value="100.0"/>
          
          <remap from="~torque_offset" to="cartesian/torque_offset"/>
          
          <param name="mu" value="0.1"/>
          
          <param name="log" type="bool" value="true"/>
                   
          <param name="is_model_floating_base" value="true"/>
          <param name="model_type" value="RBDL"/>
          
          <param name="links" type="yaml" 
                 value="[l_sole, r_sole, l_ball_tip, r_ball_tip]"/>
          
    </node>
    
    <node pkg="tf" type="static_transform_publisher" name="world_connecter"
        args="0 0 0 0 0 0 ci/base_link world 30"/>


</launch>
